clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,239 @@
+use std::{
+    env, fs,
+    io::{self, Read},
+    process::{Command, Stdio},
+};
+
+use serde_json::json;
+
+use crate::config::TeliodDaemonConfig;
+
+struct PathConfig;
+
+impl PathConfig {
+    const APP_TMP_DIR: &'static str = "/tmp/nordsecuritymeshnet";
+    const QPKG_DIR: &'static str = "/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet";
+
+    pub fn teliod_bin() -> String {
+        format!("{}/teliod", Self::QPKG_DIR)
+    }
+
+    pub fn teliod_cfg() -> String {
+        format!("{}/teliod.cfg", Self::QPKG_DIR)
+    }
+
+    pub fn teliod_log() -> String {
+        "/var/log/teliod.log".to_owned()
+    }
+
+    pub fn meshnet_log() -> String {
+        format!("{}/meshnet.log", Self::QPKG_DIR)
+    }
+
+    pub fn pid() -> String {
+        format!("{}/teliod.pid", Self::APP_TMP_DIR)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum HttpMethod {
+    Get,
+    Post,
+    Patch,
+    Unsupported,
+}
+
+impl HttpMethod {
+    fn from_method(method: &str) -> Self {
+        match method {
+            "GET" => HttpMethod::Get,
+            "POST" => HttpMethod::Post,
+            "PATCH" => HttpMethod::Patch,
+            _ => HttpMethod::Unsupported,
+        }
+    }
+
+    fn get_post_data(&self) -> String {
+        let mut data = String::new();
+        if matches!(self, HttpMethod::Post | HttpMethod::Patch) {
+            let content_length = env::var("CONTENT_LENGTH")
+                .ok()
+                .and_then(|len| len.parse::<usize>().ok());
+            if let Some(length) = content_length {
+                let _ = io::stdin().take(length as u64).read_to_string(&mut data);
+            }
+        }
+        data
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum Action {
+    Start,
+    Stop,
+    UpdateConfig,
+    GetStatus,
+    GetTeliodLogs,
+    GetMeshnetLogs,
+    Invalid,
+}
+
```

Comment:
What is the reason of creating this intermediate enum? It looks to me that you just make two `match` statements instead of one, without any visible gains.

Reply:
It looks like that, but I thought it would be good to have something to expand on in the future.

Reply:
in the end I'll change it with the `http` crate objects as you suggested below